### PR TITLE
doc: Add missing optional to getblockfrompeer

### PR DESCRIPTION
**Pull request by maflcko**
Can be reviewed with `--word-diff-regex=. --ignore-all-space`

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Needed for #23083

**Comment by Sjors** [comment]
utACK aaaa34e34db6ff0f9ca3407bda42e7f0db887737

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -777,22 +777,23 @@ static RPCHelpMan getmempoolentry()
...
-                }},
-                RPCExamples{
-                    HelpExampleCli("getblockfrompeer", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\" 0")
-                + HelpExampleRpc("getblockfrompeer", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\" 0")
-                },
+    return RPCHelpMan{
+        "getblockfrompeer",
+        "\nAttempt to fetch block from a given peer.\n"
+        "\nWe must have the header for this block, e.g. using submitheader.\n"
+        "\nReturns {} if a block-request was successfully scheduled\n",
```
Comment (jonatack) [inline src/rpc/blockchain.cpp:784]:
```diff
-        "\nReturns {} if a block-request was successfully scheduled\n",
+        "\nReturns {} if a block-request was successfully scheduled.\n",
```

**Review by jonatack — COMMENTED** [review]
Post-merge ACK, one nit for any follow-ups.

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:784]: The trailing newline isn't needed either. Whitespace will be normalized by the doc generator.

Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:784]: Sorry, I never implemented this. Done in https://github.com/bitcoin/bitcoin/pull/23714

**Pull request merged**